drivers/sensor/xbr818/xbr818.c

```diff
@@ -0,0 +1,499 @@
+/*
+ * Copyright (c) 2024 MASSDRIVER EI (massdriver.space)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT phosense_xbr818
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(XBR818, CONFIG_SENSOR_LOG_LEVEL);
+
+#include "xbr818.h"
+#include <zephyr/drivers/sensor/xbr818.h>
+
+static int xbr818_enable_i2c(const struct device *dev)
+{
+	const struct xbr818_config *config = dev->config;
+	int ret;
+
+	if (config->i2c_en.port) {
+		ret = gpio_pin_set_dt(&config->i2c_en, 1);
+		if (ret != 0) {
+			LOG_ERR("%s: could not set i2c_en pin", dev->name);
+		}
+		k_usleep(10);
+		return ret;
+	}
+	return 0;
+}
+
+static int xbr818_disable_i2c(const struct device *dev)
+{
+	const struct xbr818_config *config = dev->config;
+	int ret;
+
+	if (config->i2c_en.port) {
+		ret = gpio_pin_set_dt(&config->i2c_en, 0);
+		if (ret != 0) {
+			LOG_ERR("%s: could not unset i2c_en pin", dev->name);
+		}
+		return ret;
+	}
+	return 0;
+}
+
+static int xbr818_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	const struct xbr818_config *config = dev->config;
+	struct xbr818_data *data = dev->data;
+	int ret;
+
+	if (chan != SENSOR_CHAN_PROX && chan != SENSOR_CHAN_ALL) {
+		LOG_ERR("%s: requesting unsupported channel %i", dev->name, chan);
+		return -ENOTSUP;
+	}
+
+	ret = gpio_pin_get_dt(&config->io_val);
+	if (ret < 0) {
+		return ret;
+	}
+	data->value = (ret == 1 ? true : false);
+
+	return 0;
+}
+
+static int xbr818_channel_get(const struct device *dev, enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	struct xbr818_data *data = dev->data;
+
+	if (chan != SENSOR_CHAN_PROX) {
+		LOG_ERR("%s: requesting unsupported channel %i", dev->name, chan);
+		return -ENOTSUP;
+	}
+
+	val->val1 = (data->value ? 1 : 0);
+	val->val2 = 0;
+
+	return 0;
+}
+
+static int xbr818_attr_set(const struct device *dev, enum sensor_channel chan,
+			   enum sensor_attribute attr, const struct sensor_value *val)
+{
+	const struct xbr818_config *config = dev->config;
+	int ret;
+	uint32_t tmp;
+	uint64_t tmpf;
+
+	if (chan != SENSOR_CHAN_PROX) {
+		LOG_ERR("%s: requesting unsupported channel %i", dev->name, chan);
+		return -ENOTSUP;
+	}
+
+	if (val->val1 < 0) {
+		return -EINVAL;
+	}
+
+	ret = xbr818_enable_i2c(dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* this cast is necessary due to the additional attributes not being part of the
+	 * initial enum and causing a build error without it.
+	 */
+	switch ((int)attr) {
+	case SENSOR_ATTR_LOWER_THRESH:
+		if (val->val1 > 0xFFFF) {
+			return -EINVAL;
+		}
+		tmp = val->val1 & 0xFF;
+		ret = i2c_reg_write_byte_dt(&config->i2c, XBR818_THRESHOLD_1, tmp);
+		if (ret != 0) {
+			break;
+		}
+		tmp = (val->val1 & 0xFF00) >> 8;
+		ret = i2c_reg_write_byte_dt(&config->i2c, XBR818_THRESHOLD_2, tmp);
```

Comment:
`XBR818_THRESHOLD_1` and `XBR818_THRESHOLD_2` are sequential registers, why not use `i2c_write_dt` and read both bytes in one shot? There's a big performance difference between `i2c_write_dt` vs 2 calls to `i2c_reg_write_byte_dt`. Same applies below.

Reply:
I am also seeing a mix of doing that (each register byte written one by one) vs burst writes of a register serie in existing code, say bme280 vs mcp9600. I dont think It's worth bothering for the performance impact (if the settings here are set more than once i'd be surprised),  so I dont think ill change it, Is this ok with you?

Reply:
@MaureenHelm is the lead on sensors, I'll let her make the call but I've seen her request the same change on other drivers. I generally prefer to keep traffic to a minimum on buses. I've seen some boards where a lot of sensors are on the same bus and the extra transactions slow down all the other drivers.